django/thunderstore/community/models/package_listing.py

```diff
@@ -93,6 +93,27 @@ def get_absolute_url(self):
             kwargs={"owner": self.package.owner.name, "name": self.package.name},
         )
 
+    def get_package_version_absolute_url(self, version):
+        return reverse(
+            "packages.version.detail",
+            kwargs={
+                "owner": self.package.owner.name,
+                "name": self.package.name,
+                "version": version,
+            },
+        )
+
+    def get_full_url(self, version=None):
+        if version:
+            return str(
+                self.community.sites.first().full_url[:-1]
```

Comment:
This looks like it might inadvertently return a `http` URL when `https` should be used.

Package URLs for are built with this in mind, for example [here](https://github.com/thunderstore-io/Thunderstore/blob/3079a006e346358ff6907e0244ebb2d814aa0763/django/thunderstore/repository/models/package.py#L226):
```python
    def get_full_url(self, site: Site):
        return "%(protocol)s%(hostname)s%(path)s" % {
            "protocol": settings.PROTOCOL,
            "hostname": site.domain,
            "path": self.get_absolute_url(),
        }
```

There's also some other URL building utilities lying around in various places, such as the [make_full_url](https://github.com/thunderstore-io/Thunderstore/blob/3079a006e346358ff6907e0244ebb2d814aa0763/django/thunderstore/core/utils.py#L73)

It could be worth unifying URL building logic into a single set of utility functions somewhere, and rely on that.